api/v1/helpers/nouns/subjects.js

```diff
@@ -194,6 +200,105 @@ function applyFilters(key, filterBy) {
 } // applyFilters
 
 /**
+ * Get the aspect information from the redis and attach it to the sample object
+ * @param  {String} aspectName - Aspect name
+ * @param  {Object} sample   - Sample object
+ * @returns {Promise} - which resolves to a sample object with its aspect
+ * information
+ */
+function getAspectFromRedis(aspectName) {
+  const aspectKey = 'refocache:aspects:' + aspectName.toLowerCase();
+  return redisClient.getAsync(aspectKey)
+    .then((aspect) => {
+      if (aspect) {
+        console.log('3 this is the aspect-----' + aspect);
+        return aspect;
+      }
+      return aspect;
+    });
+} // getAspectFromRedis
+
+/**
+ * Get the sample information from Redis
+ * @param  {String} sampleKey - Sample key
+ * @returns {Promise} - which resolves to a sample object
+ */
+function getSampleFromRedis(sampleKey) {
+  let sample;
+  return redisClient.hgetallAsync(sampleKey)
+    .then((samp) => {
+      samp.relatedLinks = JSON.parse(samp.relatedLinks);
+      sample = samp;
+      console.log('2' + sample);
+      return sample;
+    });
+
+} // getSampleFromReids
+
+/**
+ *
+ * Prune a node by applying filters to it. FOR NOW NO FILTERING IS DONE.
+ *
+ * For each of the subject passed to this function, it gets the
+ * corresponding samples if any from Redis and for each of the samples, the
+ * corresponding aspect object is attached to it.
+ *
+ * @param {Object} res - Node to be pruned
+ * @returns {Integer} - Returns zero only if the filters is set and all the
+ *  samples are filtered.
+ */
+function pruneNodeV2(res) {
+  res.samples = [];
+  const subjectKey = SUBJECT_SET+SEP+res.absolutePath.toLowerCase();
+  return redisClient.smembersAsync(subjectKey)
+    .then((aspectNames) => {
+      if (aspectNames && aspectNames.length) {
+        aspectNames.forEach((aspect) => {
+          console.log('1')
+          const sampleKey = SAMPLE_SET + SEP + res.absolutePath.toLowerCase() +
+           '|' + aspect;
+          const sample = getSampleFromRedis(sampleKey);
+          const aspectObj = getAspectFromRedis(aspect);
+          let a;
+          Promise.join(sample, aspectObj, () => {
+            a = JSON.parse(aspectObj.value());
```

Comment:
Got the aspect issue fixed. I still have to clean up a lot. 
More importantly, I used the bluebird "join promise" feature to fix it. 
@iamigo @pallavi2209